app/models/payment.rb

```diff
@@ -5,5 +5,8 @@
 # "amount" is in integer quantity in halalas, or hundredths of a riyal.
 
 class Payment < ActiveRecord::Base
+  belongs_to :source, class_name: 'Cashbox'
+  belongs_to :destination, class_name: 'Cashbox'
+
   validates :amount, numericality: { only_integer: true }, allow_nil: false
```

Comment:
With the deposits/withdrawals implementation, the amount should never be negative, correct? If so, we should add validation for that